source/time_stepping/conduction_time_step.cc

```diff
@@ -0,0 +1,128 @@
+/*
+  Copyright (C) 2018 - 2020 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/global.h>
+#include <aspect/time_stepping/conduction_time_step.h>
+#include <aspect/simulator.h>
+
+namespace aspect
+{
+  namespace TimeStepping
+  {
+    template <int dim>
+    std::pair<Reaction, double>
+    ConductionTimeStep<dim>::execute()
+    {
+      double min_local_conduction_timestep = std::numeric_limits<double>::max();
+
+      const QIterated<dim> quadrature_formula (QTrapez<1>(),
+                                               this->get_parameters().stokes_velocity_degree);
+
+      FEValues<dim> fe_values (this->get_mapping(),
+                               this->get_fe(),
+                               quadrature_formula,
+                               update_values |
+                               update_gradients |
+                               update_quadrature_points);
+
+      const unsigned int n_q_points = quadrature_formula.size();
+
+      MaterialModel::MaterialModelInputs<dim> in(n_q_points,
+                                                 this->introspection().n_compositional_fields);
+      MaterialModel::MaterialModelOutputs<dim> out(n_q_points,
+                                                   this->introspection().n_compositional_fields);
+
+      for (const auto &cell : this->get_dof_handler().active_cell_iterators())
+        if (cell->is_locally_owned())
+          {
+            fe_values.reinit (cell);
+            in.reinit(fe_values,
+                      cell,
+                      this->introspection(),
+                      this->get_solution());
+
+            this->get_material_model().evaluate(in, out);
+
+            if (this->get_parameters().formulation_temperature_equation
```

Comment:
why did you decide to go for if and else statements and not a switch and case system? The switch case systems seems to me to describe better what you are doing here, but that might be a personal preference.

Reply:
No specific reason (this code has just been moved around), but a quick search says that we do it exactly in the same way in all other 5+ places in the code.
A switch is probably a few more lines, but I don't mind making that change (but better in a a separate PR?).